Mathlib/LinearAlgebra/InvariantBasisNumber.lean

```diff
@@ -302,20 +304,28 @@ private def induced_equiv [Fintype ι'] (I : Ideal R) (e : (ι → R) ≃ₗ[R]
 
 end
 
-section
-
-attribute [local instance] Ideal.Quotient.field
-
-/-- Nontrivial commutative rings have the invariant basis number property.
+/-- Nontrivial commutative semirings satisfy the rank condition.
 
 In fact, any nontrivial commutative ring satisfies the strong rank condition, see
-`commRing_strongRankCondition`. We prove this instance separately to avoid dependency on
-`LinearAlgebra.Charpoly.Basic`. -/
-instance (priority := 100) invariantBasisNumber_of_nontrivial_of_commRing {R : Type u} [CommRing R]
-    [Nontrivial R] : InvariantBasisNumber R :=
-  ⟨fun e =>
-    let ⟨I, _hI⟩ := Ideal.exists_maximal R
-    eq_of_fin_equiv (R ⧸ I)
-      ((Ideal.piQuotEquiv _ _).symm ≪≫ₗ induced_equiv _ e ≪≫ₗ Ideal.piQuotEquiv _ _)⟩
-
-end
+`commRing_strongRankCondition`, but it is not clear whether this generalizes to semirings.
+We prove this instance separately to avoid dependency on `LinearAlgebra.Charpoly.Basic`. -/
```

Comment:
```suggestion

Even if it does, we still want to prove this instance separately to avoid a dependency on `LinearAlgebra.Charpoly.Basic`. -/
```

Reply:
If we need to create a new file for the new result, we would just add a cross reference here to the new result.

Reply:
Instead of creating a new file, I moved the result to a file that already imported `Matrix.toLin`, but this PR still has a large-import warning:
`+2.05%	Mathlib.LinearAlgebra.Matrix.InvariantBasisNumber`
I can inline the proof of `Projective.of_basis` to remove the dependency on Mathlib.Algebra.Module.Projective, which will cut the imports increase from 23 to 8; should I do that?